src/helper/addFiltration.ts

```diff
@@ -0,0 +1,26 @@
+import { Request } from "express";
+import { SelectQueryBuilder } from "kysely";
+import {  TableName } from "../database";
+import { DB } from "../database";
+import { addProductFilters } from "../helper/addProductFilters";
+import { addCategoryFilters } from "./addCategoryFilters";
+
+
+export function addFiltration(
+  table: TableName,
+  query: SelectQueryBuilder<DB, TableName, any>,
+  req: Request,
+) {
+
+  if (table === "product") {
+    query = addProductFilters(req, query as any);
+  }
+   else if(table === "product_category"){
```

Comment:
"product_category" declare it as constant if intend to use it again somewhere in the codebase